### PR TITLE
Update: Column config reordering, sorting ability

### LINKED ISSUE #985: Add metadata to indicate whether or not a column supports sorting
Yea, I didn't have the answers to some of the datasource specific issues, so I mostly left them in their original condition. Maybe this is something we can attach to our metadata models to see which columns support sorting and then statically configure for now

_Originally posted by @kevinhinterlong in https://github.com/yahoo/navi/pull/983#discussion_r482664466_

### LINKED ISSUE #996: Add sort setting to column config

### PR DESCRIPTION
**Pull request by kevinhinterlong**
Resolves #996 
Resolves #985

## Description
- adds metadata to columns indicating whether or not they support sorting, (e.g. false for fili dimensions)
- ability to add/remove sort from column config
- update table with column metadata for sorting

## Screenshots
sort-inactive sort-active

https://user-images.githubusercontent.com/12093492/110698966-ebcad600-81b3-11eb-9721-fc77fe82d0a6.mp4

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/components/navi-column-config.hbs [hunk 1]
```diff
@@ -15,11 +15,13 @@
...
             @cloneColumn={{fn this.cloneColumn column}}
             @onAddFilter={{fn this.onAddFilter column.fragment}}
+            @onUpsertSort={{fn @onUpsertSort column.fragment}}
+            @onRemoveSort={{fn @onRemoveSort column.fragment}}
             @onRenameColumn={{fn @onRenameColumn column.fragment}}
             @onUpdateColumnParam={{fn @onUpdateColumnParam column.fragment}}
           />
         {{/each}}
-      </ul>
+      </ol>
```
Comment (jkusa) [inline packages/reports/addon/components/navi-column-config.hbs:24]: 👍

File: packages/reports/addon/components/navi-column-config.hbs [hunk 2]
```diff
@@ -15,11 +15,13 @@
             @onRemoveColumn={{fn @onRemoveColumn column.fragment}}
             @cloneColumn={{fn this.cloneColumn column}}
             @onAddFilter={{fn this.onAddFilter column.fragment}}
+            @onUpsertSort={{fn @onUpsertSort column.fragment}}
```
Comment (jkusa) [inline packages/reports/addon/components/navi-column-config.hbs:18]: Do we have a way for users to control the sort order?

File: packages/reports/tests/acceptance/fili-datasource-test.ts [hunk 3]
```diff
@@ -147,4 +147,25 @@ module('Acceptance | fili datasource', function (hooks) {
...
   });
+
+  test('Fili Dimension sorting disabled', async function (assert) {
+    assert.expect(4);
+    await visit('/reports/new');
+
+    await clickItem('dimension', 'Date Time');
+    assert
+      .dom('.navi-column-config-base__sort-icon')
+      .doesNotHaveAttribute('disabled', 'A timeDimension column can be sorted');
```
Comment (jkusa) [inline packages/reports/tests/acceptance/fili-datasource-test.ts:158]: will `.isNotDisabled(...)` work here?

Comment (kevinhinterlong) [reply packages/reports/tests/acceptance/fili-datasource-test.ts:158]: Nope, tried that unfortunately it checks the type and it has to match one of these

https://github.com/simplabs/qunit-dom/blob/b53ab2614337560d549c094816396ae66163fe7f/lib/assertions/is-disabled.ts#L9-L15

File: packages/reports/addon/components/navi-column-config.hbs (hunk 2 above)
Comment (kevinhinterlong) [reply packages/reports/addon/components/navi-column-config.hbs:18]: https://github.com/yavin-dev/framework/issues/1286

File: packages/core/addon/templates/components/table-renderer.hbs [hunk 4]
```diff
@@ -20,43 +20,41 @@
...
-                @duration={{2000}}
-                @enableLazyRendering={{true}}
-              />
-            {{/if}}
-          </group.item>
-        {{/each}}
-      </SortableGroup>
+              <span
+                class="table-header-cell__title {{if column.fragment.alias "table-header-cell__title--custom-name"}}"
+                title="Drag column header to reorder"
```
[the remark this answers is not in the document]
Comment (kevinhinterlong) [reply packages/core/addon/templates/components/table-renderer.hbs:35]: https://github.com/yavin-dev/framework/issues/1287

File: packages/reports/addon/components/navi-column-config/item.hbs [hunk 5]
```diff
@@ -14,7 +16,7 @@
       disabled={{@column.isRequired}}
       {{on "click" @onRemoveColumn}}
     >
-      <NaviIcon @icon="minus-circle" />
+      <DenaliIcon @icon="minus-circle" @size="extrasmall" />
```
Comment (jkusa) [inline packages/reports/addon/components/navi-column-config/item.hbs:19]: Can we make this 14px?

**Review by jkusa — COMMENTED** [review]
Overall looks good. A few things that caught my eye:
- A dragged column is transparent, we should either make the background opaque or have a shade with opacity < 100
- Did the font size change to the visualization and filter headers on purpose

**Comment by kevinhinterlong** [comment]
> Overall looks good. A few things that caught my eye:
> 
> * A dragged column is transparent, we should either make the background opaque or have a shade with opacity < 100
> * Did the font size change to the visualization and filter headers on purpose

1. 👍  added some opacity
2. Not intentional, updated

**Comment by kevinhinterlong** [comment]
https://user-images.githubusercontent.com/12093492/111505539-b889c900-8716-11eb-86b2-f6ac3d4c7bbc.mp4

File: packages/reports/app/styles/navi-reports/components/navi-column-config/base.scss [hunk 6]
```diff
@@ -8,9 +8,7 @@
   margin-top: -10px;
 
   &__api-column-name {
-    background: $color-grey-300;
-    border-radius: 4px;
-    color: $color-grey-800;
+    // border-radius: 4px;
```
Comment (jkusa) [inline packages/reports/app/styles/navi-reports/components/navi-column-config/base.scss:11]:
```suggestion
```

**Review by jkusa — APPROVED** [review]
Just the one comment, otherwise 👍

**Pull request merged**